x/mongo/driver/driver.go

```diff
@@ -65,6 +65,21 @@ type Expirable interface {
 	Alive() bool
 }
 
+// Streamer represents a Connection that supports streaming wire protocol messages using the moreToCome and
+// exhaustAllowed flags.
+//
+// The SetStreaming and CurrentlyStreaming functions correspond to the moreToCome flag on server responses. If a
+// response has moreToCome set, SetStreaming(true) will be called and CurrentlyStreaming() should return true.
+//
+// CanStream corresponds to the exhaustAllowed flag. The operations layer will set exhaustAllowed on outgoing wire
+// messages to inform the server that the driver supports streaming.
+type Streamer interface {
+	Connection
+	SetStreaming(bool)
```

Comment:
I opted to give these functions names that define the semantics of `exhaustAllowed`/`moreToCome`. If this is confusing, I'm open to giving them more literal names like `SetMoreToCome`, `IsMoreToCome`, and `SupportsExhaustAllowed`.